matchzoo/contrib/models/lbmp.py

```diff
@@ -0,0 +1,131 @@
+        >>> model.params['embedding_output_dim'] = 100
+        >>> model.params['embedding_trainable'] = True
+        >>> model.params['leaky_relu_alpha'] = 0.1
+        >>> model.params['mlp_num_layers'] = 3
+        >>> model.params['mlp_num_units'] = [600, 600]
+        >>> model.params['lstm_num_units'] = 600
+        >>> model.params['dropout_rate'] = 0.1
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params(with_embedding=True)
+        params['optimizer'] = 'adam'
+        params.add(Param(name='leaky_relu_alpha', value=0.1,
+                         desc="Negative slope coefficient of LeakyReLU function."))
+        params.add(Param(name='mlp_num_layers', value=3,
+                         desc="The number of layers of mlp."))
+        params.add(Param(name='mlp_num_units', value=[600, 600],
+                         desc="The hidden size of the FC layers, but not include the final layer."))
+        params.add(Param(name='lstm_num_units', value=600,
+                         desc="The hidden size of the LSTM layer."))
+        params.add(Param(
+            name='dropout_rate', value=0.1,
+            hyper_space=hyper_spaces.quniform(
+                low=0.0, high=0.8, q=0.01),
+            desc="The dropout rate."
+        ))
+        return params
+
+    def build(self):
+        """Build model structure."""
+        input_left, input_right = self._make_inputs()
+
+        embedding = self._make_embedding_layer()
+        embed_left = embedding(input_left)
+        embed_right = embedding(input_right)
+      
+        # get sentence embedding
+        embed_sen_left = self._sentence_encoder(
+            embed_left, lstm_num_units=self._params['lstm_num_units'], drop_rate=self._params['dropout_rate'])
+        embed_sen_right = self._sentence_encoder(
+            embed_right, lstm_num_units=self._params['lstm_num_units'], drop_rate=self._params['dropout_rate'])
+
+        # concatenate two sentence embedding: [embed_sen_left, embed_sen_right, |embed_sen_left-embed_sen_right|, embed_sen_left*embed_sen_right]
+        concat1 = keras.layers.Subtract()([embed_sen_left, embed_sen_right])
+        concat1_abs = keras.layers.Lambda(lambda x: abs(x))(concat1)
+        concat2 = keras.layers.Multiply()([embed_sen_left, embed_sen_right])
+        concat = keras.layers.Concatenate(axis=1)(
+            [embed_sen_left, embed_sen_right, concat1_abs, concat2])
+
+        # multiply perception layers to classify
+        out = self._classifier(concat, mlp_num_layers=self._params['mlp_num_layers'], mlp_num_units=self._params[
+                               'mlp_num_units'], drop_rate=self._params['dropout_rate'], leaky_relu_alpah=self._params['leaky_relu_alpha'])
+
+        self._backend = keras.Model(
+            inputs=[input_left, input_right], outputs=out)
+
+    def _classifier(
```

Comment:
you don't need to write any documentation and type hints for private methods, this saves quite a lot of time.